filters/include/pcl/filters/impl/crop_hull.hpp

```diff
@@ -44,31 +44,20 @@
 template<typename PointT> void
 pcl::CropHull<PointT>::applyFilter (PointCloud &output)
 {
-  if (dim_ == 2)
-  {
-    // in this case we are assuming all the points lie in the same plane as the
-    // 2D convex hull, so the choice of projection just changes the
-    // conditioning of the problem: choose to squash the XYZ component of the
-    // hull-points that has least variation - this will also give reasonable
-    // results if the points don't lie exactly in the same plane
-    const Eigen::Vector3f range = getHullCloudRange ();
-    if (range[0] <= range[1] && range[0] <= range[2])
-      applyFilter2D<1,2> (output);
-    else if (range[1] <= range[2] && range[1] <= range[0])
-      applyFilter2D<2,0> (output);
-    else
-      applyFilter2D<0,1> (output);
-  }
-  else
-  {
-    applyFilter3D (output);
-  }
+  output.clear();
+  std::vector<int> indices;
+  applyFilter(indices);
+  pcl::copyPointCloud(*input_, indices, output);
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template<typename PointT> void
 pcl::CropHull<PointT>::applyFilter (std::vector<int> &indices)
 {
+  indices.clear();
+  indices.reserve(indices_->size());
+
+  //std::min_element(range.data(), range.data() + 3);
```

Comment:
No need to add a commented out code-block

Reply:
fix